### PR TITLE
HMAN-171 - Error if hearingVenue.locationReferences.key != to EPIMS

### PR DESCRIPTION
**Pull request by sullivangareth**
### JIRA link (if applicable) ###

https://tools.hmcts.net/jira/browse/HMAN-171

### Change description ###

**Problem:** If ListAssist do not populate the hearing response with a hearingVenue.locationReferences.key = EPIMS then by our spec we don't populate any value in the database for hearingDayDetails.venueId. This however gives a db error

**Solution:** If the locationReferences array does NOT contain a key - "EPIMS" , then throw a validation error with appropriate message such as "Location reference EPIMS must be supplied". 

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[X] No
```

### CODE REVIEW

File: config/owasp/suppressions.xml [hunk 1]
```diff
@@ -12,6 +12,10 @@
     <cve>CVE-2013-4152</cve>
     <cve>CVE-2014-0054</cve>
     <cve>CVE-2013-7315</cve>
+    <cve>CVE-2022-22968</cve>
+    <cve>CVE-2022-22968</cve>
+    <cve>CVE-2020-11022</cve>
+    <cve>CVE-2020-11023</cve>
```
Comment (danlysiak) [inline config/owasp/suppressions.xml:18]: Should be able to revert these changes now as master has been updated.

Comment (danlysiak) [inline config/owasp/suppressions.xml:18]: Should be able to revert these changes now as master has been updated.

File: src/main/java/uk/gov/hmcts/reform/hmc/constants/Constants.java [hunk 2]
```diff
@@ -10,4 +10,5 @@ private Constants() {
         + "your browser or try again later";
     public static final String INVALID_ERROR_CODE_ERR_MESSAGE = "Error code is invalid";
     public static final String INVALID_HEARING_PAYLOAD = "Invalid json request";
+    public static final String INVALID_LOCATION_REFERENCES = "Location reference EPIMS must be supplied";
```
Comment (danlysiak) [inline src/main/java/uk/gov/hmcts/reform/hmc/constants/Constants.java:13]: Maybe..

```suggestion
    public static final String INVALID_LOCATION_REFERENCES = "At least one EPIMS location reference must be supplied";
```

**Review by danlysiak — APPROVED** [review]
Couple of minor comments but not about the implementation, so ✅

File: src/main/java/uk/gov/hmcts/reform/hmc/constants/Constants.java (hunk 2 above)
Comment (sullivangareth) [reply src/main/java/uk/gov/hmcts/reform/hmc/constants/Constants.java]: Fixed in https://github.com/hmcts/hmc-hmi-inbound-adapter/pull/57/commits/d6c7102ac5ec829df9b005bb47e9e623bb26f9f9

File: config/owasp/suppressions.xml (hunk 1 above)
Comment (sullivangareth) [reply config/owasp/suppressions.xml]: fixed in https://github.com/hmcts/hmc-hmi-inbound-adapter/pull/57/commits/d6c7102ac5ec829df9b005bb47e9e623bb26f9f9

**Pull request merged**